src/libertem/job/sum.py

```diff
@@ -9,25 +9,21 @@ def get_tasks(self):
             yield SumFramesTask(partition=partition)
 
     def get_result_shape(self):
-        return self.dataset.shape[2:]
+        return self.dataset.shape.sig
 
 
 class SumFramesTask(Task):
     def __call__(self):
         """
         sum frames
         """
-        part = np.zeros(self.partition.dataset.shape[2:], dtype="float32")
+        part = np.zeros(self.partition.dataset.shape.sig, dtype="float32")
         for data_tile in self.partition.get_tiles():
             data = data_tile.data
             if data.dtype.kind == 'u':
                 data = data.astype("float32")
             result = data_tile.data.sum(axis=(0, 1))
-            tile_slice = data_tile.tile_slice.get()
-            part[
-                tile_slice[2],
-                tile_slice[3],
-            ] += result
+            part[data_tile.tile_slice.get(sig_only=True)] += result
```

Comment:
A bit of documentation on how to use the Shape could be nice. This point here is extremely elegant and nice. Just not obvious to do it like that for someone using Shape. Could be a good example for documentation.

Reply:
Idea for follow-up: How does Hyperspy do this? Perhaps comparison or improvement?